### PR TITLE
Set max width for icons in the quick open popup

### LINKED ISSUE #93690: Custom SVG icon appears too large in Quick Open (Ctrl+P)
### Tested versions

v4.3.beta2.official [b75f0485b]

### System information

macOS Sonoma 14.1.1 - OpenGL API 4.1 Metal - 88 - Compatibility - Using Device: Apple - Apple M1 Pro

### Issue description

The custom icon for resources appears too large in Quick Open.

Given a custom resource like this:

```gdscript
@icon("res://icon.svg")

class_name MyResource
extends Resource
```

And a `.tres` file created with the custom resource script like this:
Screenshot 2024-06-28 at 2 09 02 AM

When bringing up `Quick Open`, it looks like this:
Screenshot 2024-06-28 at 2 04 59 AM

### Steps to reproduce

1. Create a fresh project
1. Create a custom resource script
1. Set a custom icon for the custom resource, use the default svg that comes with the project
1. Create a custom resource file `.tres` using the custom resource script created in the prev step
1. Close and Open the editor for the icon to be visible in Inspector and Quick Open (this is a separate bug)
1. Bring up `Quick Open` (Cmd+P in macos)
1. Observe how the icon appears too large

### Minimal reproduction project (MRP)

n/a

### PR DESCRIPTION
**Pull request by poga**
This fix https://github.com/godotengine/godot/issues/93690.

### CODE REVIEW

File: editor/editor_quick_open.cpp [hunk 1]
```diff
@@ -123,6 +123,7 @@ void EditorQuickOpen::_update_search() {
 		for (int i = 0; i < entry_limit; i++) {
 			TreeItem *ti = search_options->create_item(root);
 			ti->set_text(0, entries[i].path);
+			ti->set_icon_max_width(0, 16 * EDSCALE);
```
Comment (KoBeWi) [inline editor/editor_quick_open.cpp:126]: You should use `class_icon_size` theme constant.
But don't use `get_theme_constant()` inside the loop.

Comment (poga) [reply editor/editor_quick_open.cpp]: Thanks for the hint! But I'm not sure if have to multiply it by `EDSCALE`

Comment (KoBeWi) [reply editor/editor_quick_open.cpp]: It shouldn't be multiplied.

File: editor/editor_quick_open.cpp [hunk 2]
```diff
@@ -119,10 +119,12 @@ void EditorQuickOpen::_update_search() {
 			sorter.sort(entries.ptrw(), entries.size());
 		}
 
+		int class_icon_size = search_options->get_theme_constant(SNAME("class_icon_size"), SNAME("Editor"));
```
Comment (KoBeWi) [inline editor/editor_quick_open.cpp:122]:
```suggestion
		const int class_icon_size = search_options->get_theme_constant(SNAME("class_icon_size"), EditorStringName(Editor));
```

**Review by KoBeWi — APPROVED** [review]
The commits need to be squashed.

**Comment by poga** [comment]
Squashed!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by z4gon** [comment]
Thank you for the fix!

**Pull request merged**